### PR TITLE
Tabular: Remove TabularPrediction

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Removed TabularPrediction
- Updated dataset -> data for TabularPredictor method arguments
- Updated as_pandas=True by default
- Added attributes documentation

Error when TabularPrediction.fit() is called:

```
AssertionError: CRITICAL: TabularPrediction has been removed and has been replaced by TabularPredictor.
Please use `autogluon.tabular.TabularPredictor` instead and refer to documentation for tutorials:
https://auto.gluon.ai/dev/tutorials/tabular_prediction/tabular-quickstart.html 
https://auto.gluon.ai/dev/tutorials/tabular_prediction/tabular-indepth.html 
https://auto.gluon.ai/dev/api/autogluon.task.html#autogluon.tabular.TabularPredictor 
https://auto.gluon.ai/dev/api/autogluon.task.html#autogluon.tabular.TabularPredictor.fit 
https://auto.gluon.ai/dev/api/autogluon.task.html#tabulardataset
```

TabularPrediction will be fully removed nearer to v0.1 release.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-892-3 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-892/3/index.html

File: docs/tutorials/tabular_prediction/tabular-kaggle.md [hunk 1]
```diff
@@ -73,25 +73,25 @@ test_transaction = pd.read_csv(directory+'test_transaction.csv')
...
-predictor.class_labels # classes in this list correspond to columns of predict_proba() output
+predictor.class_labels  # classes in this list correspond to columns of predict_proba() output
 ```
 
 Alternatively, the following command should clarify which predicted-probability corresponds to which class:
 
 ```
-y_predproba = predictor.predict_proba(test_data, as_pandas=True)
+y_predproba = predictor.predict_proba(test_data)
 ```
```
Comment (jwmueller) [inline docs/tutorials/tabular_prediction/tabular-kaggle.md:95]: can delete this now that `as_pandas` is default right?

Comment (Innixma) [reply docs/tutorials/tabular_prediction/tabular-kaggle.md:95]: probably, although we can look at these little details closer to release.

**Review by jwmueller — APPROVED** [review]
Looks good. When will we get rid of the TabularPredictorV1?

**Comment by Innixma** [comment]
> Looks good. When will we get rid of the TabularPredictorV1?

Probably in the next couple days. The only thing that depended on it was TabularPrediction. All I need to do is move the methods to TabularPredictor.

**Pull request merged**